Feature/123 demo

MichaelXian opened the pull request:
The ResultData object now matches the database schema (except id,) and ResultsRow reflects the changes, but doesn't display a large portion of the data. 
Should the id be included in ResultsData?
Do we want to display the rest of the data for the demo?

cowmanjoe committed:
Got client to server search functionality working, pretty hacky though.

cormacjmollitor committed:
Able to Search For Jobs for Demo

The navbar is now properly hooked up to the backend to display returned postings from search queries. It doesn't use Redux so it's hacky for now but it's sufficient for our demo

cormacjmollitor committed:
Now redirects to results table on search

MichaelXian committed:
Make ResultsTable match database schema

MichaelXian committed:
Make ResultRow reflect the change in ResultData

MichaelXian committed:
Fix typo in ResultData and postingsReducer

MichaelXian committed:
Remove location from ResultData

cowmanjoe commented on client/src/reducers/postingsReducer.js:
```diff
@@ -3,9 +3,46 @@ import ResultData from '../components/ResultData';
+    "0.000",
+    "0.000",
+    "Facebook",
+    "5",
```
I don't think redux is being used anymore in our client project. In feature/121 the redux stuff is commented out so it actually doesn't do anything. Also I'm not sure that we should have dummy data anymore since the front end is able to hit the backend. I'm not sure about that last point though because it may make front end development easier. Regardless, I don't think the dummy data should be here.

cowmanjoe replied:
If you are keeping t he dummy data, I think the longitude, latitude, and salaryMin should be numbers, not strings.

MichaelXian replied:
Should I just remove the entire file then?

cowmanjoe commented on client/src/components/ResultData.js:
```diff
@@ -1,10 +1,15 @@
-  constructor(title, description, excerpt, location, postUrl, applyUrl) {
-    this.title = title;
+  constructor(job_title, link, description, city, state, country, latitude, longitude, company_name, salary_min, start_date) {
+    this.job_title = job_title;
```
I think it makes sense to have id in here, just in case it could be used in the future for sending out requests e.g. `GET/job/:id`.

MichaelXian replied:
Added id

cowmanjoe commented on client/src/components/ResultRow.js:
```diff
@@ -26,22 +26,22 @@ class ResultRow extends Component {
     let originalPosting = '';
     let applyButton = '';
-    let { city } = this.props.result;
+    let location = this.props.result.city + ", " + this.props.result.state + " " + this.props.result.country;
```
Shouldn't there be a comma between state and country as well?

MichaelXian replied:
Added the comma. Also changed it to use a template literal for the location string.

cowmanjoe reviewed (changes requested):
This is probably a good idea to merge in before I merge my changes from feature/121 into develop. Requested a few changes. One other request if you have time: can you show the company name somewhere on the result row? I think it's a very important aspect to show this.

MichaelXian committed:
Add ID to ResultData

MichaelXian committed:
Update table row

The table row will now have a comma between state and country. It now
uses a template literal for the location, rather than concatenating
strings and variables. It also shows the company name right under the
job title. Also fixed a typo in ResultData.

MichaelXian commented:
Added the company name under the job title. It doesn't look particularly good, but it exists now!

AmirNaghibi commented:
The demo has been done successfully in November 2018; so I close the ticket.
